src-v2/NTS.Domain.Core/Objects/Startlist.cs

```diff
@@ -42,26 +41,41 @@ public void AssignStarts(IEnumerable<Participation> participations)
             if (!participation.IsEliminated())
             {
                 var phases = participation.Phases;
-                var now = new Timestamp(DateTime.Now);
-                foreach (var phase in phases.Where(p => p.StartTime != null))
+                foreach (var phase in phases)
                 {
+                    if (phase.StartTime == null)
+                    {
+                        continue;
+                    }
                     var phaseIndex = phases.IndexOf(phase);
+                    var phaseNumber = phaseIndex + 1;
                     var start = new Start(
                         participation.Combination.Name,
                         participation.Combination.Number,
-                        phaseIndex + 1,
+                        phaseNumber,
                         phases[phaseIndex].Length,
-                        participation.Phases.Distance,
-                        phase.StartTime!.DateTime.TimeOfDay
+                        phase.StartTime.DateTime
                     );
                     Starts.Add(start);
                 }
             }
         }
     }
 
-    public void Add(Start start)
+    void PhaseCompletedHandler(PhaseCompleted phaseCompleted)
+    {
+        var newStart = new Start(phaseCompleted.Participation);
+        Starts.Add(newStart);
+        ChangeHandler();
+    }
+
+    TimeSpan CurrentTime()
+    {
+        return DateTimeOffset.Now.TimeOfDay;
+    }
+
+    List<Start> OrderByTimeAndPhase(IEnumerable<Start> starts)
     {
-        Starts.Add(start);
+        return starts.OrderBy(s => s.Time).OrderBy(s => s.PhaseNumber).ToList();
```

Comment:
This isn't actually going to order by time at all, because the second `OrderBy` would simply overwrite the first one. Instead you should use `OrtherBy(...).ThenBy(...)`